apps/www/registry/new-york/ui/timeline.tsx

```diff
@@ -0,0 +1,208 @@
+        {...props}
+      >
+        {children}
+      </ul>
+    )
+  }
+)
+Timeline.displayName = "Timeline"
+
+const timelineItemVariants = cva("grid items-center gap-x-2", {
+  variants: {
+    status: {
+      done: "text-primary",
+      default: "text-muted-foreground",
+    },
+  },
+  defaultVariants: {
+    status: "default",
+  },
+})
+
+interface TimelineItemProps
+  extends React.HTMLAttributes<HTMLLIElement>,
+    VariantProps<typeof timelineItemVariants> {}
+
+const TimelineItem = React.forwardRef<HTMLLIElement, TimelineItemProps>(
+  ({ className, status, ...props }, ref) => (
+    <li
+      className={cn(timelineItemVariants({ status }), className)}
+      ref={ref}
+      {...props}
+    />
+  )
+)
+TimelineItem.displayName = "TimelineItem"
+
+const timelineDotVariants = cva(
+  "col-start-2 col-end-3 row-start-1 row-end-1 flex size-4 items-center justify-center rounded-full border border-current",
+  {
+    variants: {
+      status: {
+        default: "[&>*]:hidden",
+        current:
+          "[&>*:not(.radix-circle)]:hidden [&>.radix-circle]:bg-current [&>.radix-circle]:fill-current",
+        done: "bg-primary [&>*:not(.radix-check)]:hidden [&>.radix-check]:text-background",
+        error:
+          "border-destructive bg-destructive [&>*:not(.radix-cross)]:hidden [&>.radix-cross]:text-background",
+        custom: "[&>*:not(:nth-child(4))]:hidden [&>*:nth-child(4)]:block",
+      },
+    },
+    defaultVariants: {
+      status: "default",
+    },
+  }
+)
+
+interface TimelineDotProps
+  extends React.HTMLAttributes<HTMLDivElement>,
+    VariantProps<typeof timelineDotVariants> {
+  customIcon?: React.ReactNode
+}
+
+const TimelineDot = React.forwardRef<HTMLDivElement, TimelineDotProps>(
+  ({ className, status, customIcon, ...props }, ref) => (
+    <div
+      role="status"
+      className={cn("timeline-dot", timelineDotVariants({ status }), className)}
+      ref={ref}
+      {...props}
+    >
+      <div className="radix-circle size-2.5 rounded-full" />
+      <CheckIcon className="radix-check size-3" />
+      <Cross1Icon className="radix-cross size-2.5" />
+      {customIcon}
+    </div>
+  )
+)
+TimelineDot.displayName = "TimelineDot"
+
+const timelineContentVariants = cva(
+  "row-start-2 row-end-2 pb-8 text-muted-foreground",
+  {
+    variants: {
+      side: {
+        right: "col-start-3 col-end-4 mr-auto text-left",
+        left: "col-start-1 col-end-2 ml-auto text-right",
+      },
+    },
+    defaultVariants: {
+      side: "right",
+    },
+  }
+)
+
+interface TimelineConentProps
+  extends React.HTMLAttributes<HTMLParagraphElement>,
+    VariantProps<typeof timelineContentVariants> {}
+
+const TimelineContent = React.forwardRef<
+  HTMLParagraphElement,
+  TimelineConentProps
+>(({ className, side, ...props }, ref) => (
+  <p
+    className={cn(timelineContentVariants({ side }), className)}
+    ref={ref}
+    {...props}
+  />
+))
+TimelineContent.displayName = "TimelineContent"
```

Comment:
Hey man, I just used this component in my project, thank you so much! 

I don't think timeline content should be limited to just text. In my project there was a case where I needed to pass <div> element in and it cause a hydration. Changing this <p> to <div> should have the same effect but will avoid error

Reply:
Also, it renders very strangely on mobile. Works fine on a laptop, even when I set screen size to small. I will try to debug it let you know how I get on

![IMG_5357](https://github.com/shadcn-ui/ui/assets/64111480/c079badf-18d2-4b87-9364-30475cdbcd88)

Reply:
Same issue with this too. It works fine on inspect element emulation but once viewed on an actual mobile screen, it struggles to render.

Reply:
Managed to see a fix for it. It seems like Safari on Mac and any browser on mobile would destroy the layout. I managed to fix it by putting the entire timeline component in a div that has `display: grid` on it.

Refer to my implementation below:
https://github.com/blurridge/Gihay/commit/a91df3cd846110538528ebb02b26aca48806db0a

This could be helpful @flixlix @Toamatvey. Thanks for the amazing component btw @flixlix! Love how everything is easy to setup and is modular. Hopefully this gets merged in the future.

Reply:
fixed both issues, thank you for reporting